crates/ruff/src/rules/perflint/rules/incorrect_dict_iterator.rs

```diff
@@ -0,0 +1,262 @@
+        for el in elts {
+            if let Expr::Name(ast::ExprName { id, .. }) = el {
+                // If one of the items of the tuple is not ignored, we consider the whole tuple
+                // as unignored
+                if id != "_" {
+                    return true;
+                }
+                continue;
+            }
+            if let Expr::Tuple(_) = el {
+                return is_not_ignored_tuple_or_name(el);
+            }
+        }
+    }
+    false
+}
+
+/// W8102
+pub(crate) fn incorrect_dict_iterator(checker: &mut Checker, target: &Expr, iter: &Expr) {
+    if let Expr::Call(ast::ExprCall { func, args, .. }) = iter {
+        let Expr::Attribute(ast::ExprAttribute { attr, value, ..  }) = func.as_ref() else {
+            return;
+        };
+        if attr != "items" {
+            return;
+        }
+        if !args.is_empty() {
+            return;
+        }
+        if let Expr::Tuple(ast::ExprTuple {
+            elts,
+            range: tup_range,
+            ..
+        }) = target
+        {
+            if elts.len() != 2 {
+                return;
+            }
+
+            match &elts[0] {
+                Expr::Name(ast::ExprName { id, range, .. }) => {
+                    if id == "_" {
+                        let mut diagnostic = Diagnostic::new(
+                            IncorrectDictIterator {
+                                subset: "values".to_string(),
+                            },
+                            *range,
+                        );
+                        if let Expr::Name(ast::ExprName {
+                            range: dict_range, ..
+                        }) = value.as_ref()
+                        {
+                            if checker.patch(diagnostic.kind.rule()) {
+                                let mut fix_val: &str = "";
+                                if let Expr::Name(ast::ExprName { id, .. }) = &elts[1] {
+                                    fix_val = id.as_str();
+                                }
+                                if let Expr::Tuple(ast::ExprTuple {
+                                    range: val_range, ..
+                                }) = &elts[1]
+                                {
+                                    fix_val = checker.locator.slice(*val_range);
+                                }
+                                diagnostic.set_fix(Fix::automatic_edits(
+                                    Edit::range_replacement(
+                                        ".values".to_string(),
+                                        TextRange::new(dict_range.end(), func.range().end()),
+                                    ),
+                                    [Edit::range_replacement(fix_val.to_string(), *tup_range)],
+                                ));
+                            }
+                            checker.diagnostics.push(diagnostic);
+                            return;
```

Comment:
Shouldn't this be outside the `if let Expr::Name(...` block? The same question for other branches as well.

Reply:
Depends on what we want the behaviour to be. Current implementation does not continue if it finds a violation for the first of the two args. So a case of 

```python
for _, _ in d.items():
    pass
```

Will trigger a violation only for the first arg. Do we think it should be the responsibility of this rule to check for a completely ignored `.items()` call? And if so what do we want to do with autofixes?

Reply:
This is an interesting case. `perflint` seems to be triggering for both key and value place which doesn't seem correct to me. I guess the reason someone would want to use a loop like this is to do something `n` number of times where `n` is the dictionary length. I would suggest to ignore such cases plus I don't think it's the responsibility of this rule to check it.